### PR TITLE
Use FP32 logits in MoEGate fallbacks

### PR DESCRIPTION
**Pull request by mmangkad**
## Summary

Fixes a small gap left after #29783. The prefill-CP branch could still fall back to `F.linear` and produce BF16 router logits.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #28914364446](https://github.com/sgl-project/sglang/actions/runs/28914364446)
Latest PR Test (Extra): :x: [Run #28914364314](https://github.com/sgl-project/sglang/actions/runs/28914364314)

### CODE REVIEW

**Review by b8zhong — COMMENTED** [review]
The docstring should be outdated in FI, can you verify the code

**Comment by mmangkad** [comment]
> The docstring should be outdated in FI, can you verify the code

Yes, FI accepts BF16 or FP32 bias here, so this is not a hard runtime issue and that is why your PR passed CI. But since the accuracy issue was for NVFP4, do we really want FP32 bias/logits for the unquantized BF16 path too, or should that path stay BF16?

**Comment by b8zhong** [comment]
Let's keep the bias in f32. Since the correction bias has sensitive values

**Review by b8zhong — APPROVED** [review]
Good job.

File: python/sglang/srt/models/deepseek_v2.py [hunk 1]
```diff
@@ -480,6 +480,10 @@ def forward(
             )
 
         if get_global_server_args().enable_deterministic_inference:
+            if _is_cuda:
```
Comment (Fridge003) [inline python/sglang/srt/models/deepseek_v2.py]: We shouldn't change here since deterministic inference relies on F.linear

**Review by Fridge003 — CHANGES_REQUESTED** [review]

File: python/sglang/srt/models/deepseek_v2.py (hunk 1 above)
Comment (mmangkad) [reply python/sglang/srt/models/deepseek_v2.py]: I checked that `linear_bf16_fp32` rounds back bit-exactly to `F.linear`, but the FP32 logits can still change top-k routing before rounding, so reverted

**Review by Fridge003 — APPROVED** [review]

**Comment by Fridge003** [comment]
/rerun-test test/registered/cp/test_deepseek_v3_cp_single_node.py
